hs_core/models.py

```diff
@@ -2310,6 +2310,8 @@ def sysmeta_path(cls, resource_id):
     def delete(self, using=None, keep_parents=False):
         """Delete resource along with all of its metadata and data bag."""
         from .hydroshare import hs_bagit
+        from .hydro_realtime_signal_processor import solr_delete
+        solr_delete(self)  # avoid SOLR corruption by deleting SOLR record first
```

Comment:
What happens if a resource is in SOLRQueue and a resource is deleted?  This piece should run fine, but I'm not sure what would happen in the periodic task for that now deleted resource.